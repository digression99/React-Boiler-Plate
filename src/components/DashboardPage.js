import React from 'react';

export const DashboardPage = () => {
    return (
        <div>
            Dashboard Page Content
        </div>
    );
};

export default DashboardPage;