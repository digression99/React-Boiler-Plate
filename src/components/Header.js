import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from "../actions/auth";

export const Header = ({startLogout}) => (
    <header className="header-layout">
        <div className="content-container">
            <div className="header-layout__content">
                <Link className="header-layout__title" to="/dashboard">
                    <h1>Boilerplate</h1>
                </Link>
                <button
                    className="button button--link"
                    onClick={startLogout}
                >
                    Logout
                </button>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout : () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);