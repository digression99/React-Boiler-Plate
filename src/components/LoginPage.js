import React from 'react';
import {connect} from 'react-redux';
import { startLogin } from '../actions/auth';

export class LoginPage extends React.Component {

    onButtonClick = (e) => {
        console.log('button clicked!');
        console.log('event : ', e);
        this.props.startLogin();
    };

    render = () => (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__title">Boilerplate</h1>
                <p>Tap line for app.</p>
                <button
                    className="button-layout"
                    onClick={this.onButtonClick}
                >Login with Google</button>
            </div>
        </div>
    );
}

const mapStateToProps = (state, props) => ({
});

const mapDispatchToProps = (dispatch) => ({
    startLogin : () => dispatch(startLogin())
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
