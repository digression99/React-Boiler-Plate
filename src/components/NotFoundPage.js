import React from 'react';
import { Link } from 'react-router-dom';

export const NotFoundPage = () => {
    return (
        <div>
            {/*404! - <a href="/">Go Home!</a>*/}
            404! - <Link to="/">Go Home</Link>
        </div>
    );
};

export default NotFoundPage;