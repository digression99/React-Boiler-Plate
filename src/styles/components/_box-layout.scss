.box-layout {
  align-items : center;
  background : url('/images/bg.jpg');
  background-size : cover; // always cover the container with the image.
  display : flex;
  height : 100vh; // view port height 100 percent.
  justify-content : center;
  width : 100vw; // view port width 100 percent.
}

  .box-layout__box {
    background: fade-out(white, .55);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;
  }

  .box-layout__title {
    margin : 0 0 $m-size 0;
    line-height : 1; // all text has a line height.
  }