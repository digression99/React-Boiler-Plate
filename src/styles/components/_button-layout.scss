.button-layout {
  background : $blue;
  //width : max-width;
  font-family : Helvetica, Arial, sans-serif;
  font-weight : 500;
  font-size : $font-size-large;
  line-height : 1;
  color : white;
  display : inline-block;
  padding : $m-size $xl-size $m-size $xl-size; // top right bottom left
  justify-content : center;
  text-decoration : none;
}

.button--link {
  background : none;
  border : none;
  color : white;
  padding : 0;
}

.button--secondary {
  background : #888888;
}