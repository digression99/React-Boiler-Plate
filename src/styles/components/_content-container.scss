.content-container {
  margin : 0 auto; // auto : automatically set rest of the margins.
  padding : 0 $m-size;
  max-width : 80rem;

}