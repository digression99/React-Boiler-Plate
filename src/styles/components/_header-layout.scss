.header-layout {
  background : $dark-blue;
}

.header-layout__content {
  align-items : center;
  display : flex;
  justify-content : space-between;
  padding : $s-size 0;
}

.header-layout__title {
  color : white;
  text-decoration : none; // clear default text deco (underline)
  h1 {
    margin : 0;
  }
}