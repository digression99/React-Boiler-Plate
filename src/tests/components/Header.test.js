import React from 'react';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import {Header} from '../../components/Header';
// import toJSON from 'enzyme-to-json';

test('should render Header correctly', () => {
    const wrapper = shallow(<Header startLogout={() => {} }/>);
    // expect(wrapper.find('h1').text()).toBe('Expensify');

    expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
    const onClickSpy = jest.fn();

    const wrapper = shallow(<Header startLogout={onClickSpy}/>);

    wrapper.find('button').simulate('click');
    expect(onClickSpy).toHaveBeenCalled();
});

// should call startLogout on button click
// LoginPage test file -> should call startLogin on button click