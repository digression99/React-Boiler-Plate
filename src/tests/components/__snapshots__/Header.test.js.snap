// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header-layout"
>
  <div
    className="content-container"
  >
    <div
      className="header-layout__content"
    >
      <Link
        className="header-layout__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boilerplate
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
